django/db/models/fields/related.py

```diff
@@ -1885,6 +1899,30 @@ def __init__(self, to, to_field=None, related_name=None, related_query_name=None
             # be correct until contribute_to_class is called. Refs #12190.
             to_field = to_field or (to._meta.pk and to._meta.pk.name)
 
+        if on_delete is None:
+            warnings.warn(
+                "on_delete will be a required arg for %s in Django 2.0. "
+                "Set it to models.CASCADE if you want to maintain the current default behavior. "
+                "See https://docs.djangoproject.com/en/%s/ref/models/fields/"
+                "#django.db.models.ForeignKey.on_delete" % (
+                    self.__class__.__name__,
+                    get_docs_version(),
+                ),
+                RemovedInDjango20Warning, 2)
```

Comment:
Yes, I'm also thinking that might be simplest. It makes sense that deprecation warnings occur in the public-API class, not in an internal base class. And this doesn't really save us any duplication anyway, since `ForeignKey` still has to duplicate the warning.